app/Http/Controllers/SettingsController.php

```diff
@@ -1032,9 +1057,21 @@ public function deleteDelegateSupport($topic_num,$camp_num,$delegatedTo,$remaing
      * #831
      */
     public function remove_support($topicNum,$campNum,$nickNameId){
+        $as_of_time = time();
         $support = Support::where('camp_num', $campNum)->where('topic_num', $topicNum)->where('nick_name_id', $nickNameId)->where('end', '=', 0)->first();
         $delegateNickNameId = $support->delegate_nick_name_id;
-        $this->removeSupport($topicNum,$campNum,$nickNameId,$delegateNickNameId,$support->support_order);
+        $promoteDelegate = true;
+        $ifSupportLeft = true; //default, only useful in case od direct supporter
+        if(!$delegateNickNameId){
+            $supportCount = Support::where('topic_num', $topicNum)
+             ->whereIn('nick_name_id', [$nickNameId])
+             ->whereRaw("(start < $as_of_time) and ((end = 0) or (end > $as_of_time))")
+             ->count();
```

Comment:
probably shift the below code  
```
$supportCount = Support::where('topic_num', $topicNum)
             ->whereIn('nick_name_id', [$nickNameId])
             ->whereRaw("(start < $as_of_time) and ((end = 0) or (end > $as_of_time))")
```
to function and also use it in `emailForSupportDeleted` function.